test/resources/FileBasedApps/PackageSearchApp.cs

```diff
@@ -0,0 +1,82 @@
+using System;
+using System.Net.Http;
+using System.Text.Json;
+using System.Threading.Tasks;
+
+// Complex file-based app that searches for NuGet packages and retrieves information
+class Program
+{
+    static async Task Main(string[] args)
+    {
+        Console.WriteLine("File-based app: Package Search Demo");
+        
+        string packageId = "Microsoft.NETCore.App.Ref";
+        Console.WriteLine($"Searching for package: {packageId}");
+        
+        try
+        {
+            using var httpClient = new HttpClient();
+            
+            // Search for the package using NuGet API
+            string searchUrl = $"https://api.nuget.org/v3-flatcontainer/{packageId.ToLowerInvariant()}/index.json";
```

Comment:
This package search should use the `dotnet package search` command to do the search. It should output in json. Use detailed verbosity. It should then verify that the json contains at least some expected content (e.g. specific versions are seen, there is a valid project url, etc.).

Then, the app should convert this json format into xml, then use LINQ queries over that xml object to do something interesting. The app should output a new, formatted table with information about the download counts, sorted by most downloaded, etc.

The goal is for the app to exercise a swath of .NET functionality in a single scenario test.